shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/WADODownloaderService.java

```diff
@@ -520,6 +510,54 @@ private void extractDICOMFilesFromMHTMLFile(final byte[] responseBody, final Str
 		}
 	}
 
+	/**
+	 * This method reads in a file in format MHTML, one representation of a multipart/related response, that is given from
+	 * a PACS server, that supports WADO-RS requests.
+	 *
+	 * MHTML, short for MIME Encapsulation of Aggregate HTML Documents, is a web page archive format used to combine in a single document
+	 * the HTML code and its companion resources that are otherwise represented by external links (such as images, Flash animations, Java applets,
+	 * and audio files). The content of an MHTML file is encoded as if it were an HTML e-mail message, using the MIME type multipart/related.
+	 *
+	 * @param responseBody
+	 * @param instanceUID
+	 * @throws FileNotFoundException
+	 * @throws IOException
+	 * @throws MessagingException
+	 */
+	private void extractDICOMZipFromMHTMLFile(final byte[] responseBody, final String instanceUID, String name, ZipOutputStream zipOutputStream)
+			throws IOException, MessagingException {
+		try(ByteArrayInputStream bIS = new ByteArrayInputStream(responseBody)) {
+			ByteArrayDataSource datasource = new ByteArrayDataSource(bIS, CONTENT_TYPE_MULTIPART);
+			MimeMultipart multipart = new MimeMultipart(datasource);
+			int count = multipart.getCount();
+			if (count == 1) {
```

Comment:
Hi @jcomedouteau, maybe we could always use a for loop? in case count == 1, we loop only once?